CS3243_P1_XX_1.py

```diff
@@ -0,0 +1,201 @@
+import os
+import sys
+import copy
+import time
+
+class Puzzle(object):
+    def __init__(self, init_state, goal_state):
+        # you may add more attributes if you think is useful
+        self.init_state = init_state
+        self.goal_state = goal_state
+        self.actions = list()
+
+    def solve(self):
+        #consider flattening array into str for faster processing
+        tick = time.time()
+        row_len = len(init_state)
+
+        #one time check for solvability
+        if( (self.inversion_check(init_state,row_len)) % 2):
+            print("UNSOLVABLE")
+            return self.actions
+        #finding index of 0
+        for r in range(row_len):
+            for c in range(row_len):
+                if(init_state[r][c] == 0):
+                    index = (r,c)
+                    break
+
+        visited_nodes = {0}
+        frontier = list()
+        #frontier will hold the state of a node, the index of 0 and the list of actions to get there
+        frontier.append((init_state,index,self.actions))
+
+        while len(frontier):
+                state = frontier.pop(0)
```

Comment:
I think the indentation level for this should be 8 instead of 12?

Reply:
Oh right thanks!